src/keyDerivation.c

```diff
@@ -0,0 +1,80 @@
+#include <os_io_seproxyhal.h>
+#include <stdint.h>
+
+#include "assert.h"
+#include "errors.h"
+#include "keyDerivation.h"
+
+#define VALIDATE_PARAM(cond) if (!(cond)) THROW(ERR_INVALID_REQUEST_PARAMETERS)
+
+void validatePath(uint32_t* path, uint32_t pathLength)
+{
+	uint32_t bip44 = BIP_44 | HARDENED_BIP32;
+	uint32_t adaCoinType = ADA_COIN_TYPE | HARDENED_BIP32;
+
+	VALIDATE_PARAM(pathLength >= 3 && pathLength <= 10);
+
+	VALIDATE_PARAM(path[0] ==  bip44);
+	VALIDATE_PARAM(path[1] ==  adaCoinType);
+	VALIDATE_PARAM(path[2] >= HARDENED_BIP32);
+}
+
+void derivePrivateKey(
+        uint32_t* bip32Path, uint32_t pathLength,
+        chain_code_t* chainCode,
+        privateKey_t* privateKey
+)
+{
+	validatePath(bip32Path, pathLength);
+
+	uint8_t privateKeyRaw[64];
+
+	STATIC_ASSERT(CX_APILEVEL >= 5, unsupported_api_level);
```

Comment:
hmm, which call this static assert relates to? I guess os_perso. We should move it next to it then ...